torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -109,6 +109,9 @@ void fuseBroadcast(Block* b) {
     }
 
     for (size_t position : broadcast_positions) {
+      if (position >= n->inputs().size()) {
```

Comment:
In general I would recommend asserting that `position` is valid. It should be expected that `getBroadcastPositions` does not return invalid results. So the fix to skip optional input for `onnx::Gemm`(as a matter of fact, any operator with optinal inputs) should go into `getBroadcastPositions`.